### ISSUE TITLE
HedgeModManager cannot download and set Wine to run with

### ISSUE THREAD

**Issue by McSkellyHo** [sonic2kk/steamtinkerlaunch#662]
## System Information
- SteamTinkerLaunch version: v11.12.20221115-5
- Distribution: SteamOS
- Installation Method: Manual

## Issue Description
After using `steamtinkerlaunch hmm start nightly` to install HedgeModManager, the app doesn't launch after the popup "Starting HedgeModManager"
I checked the attached log and found two errors, both of which are `ERROR - setModWine - HMMWINE was not found - can't continue`. The second log file you mentioned, `installDotNet.logHMM`, isn't present in `/dev/shm/steamtinkerlaunch/`, I'm guessing because of the wine issue.

## Logs
[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10026272/steamtinkerlaunch.log)

**Comment by sonic2kk** [comment]
Thanks!

> I checked the attached log and found two errors, both of which are `ERROR - setModWine - HMMWINE was not found - can't continue`

This is indeed the problem, and why there is no dotnet log file. I am not totally sure why it cannot get the HMM wine. I wonder if the GitHub downloading is broken...

Just to make sure:
1. You installed SteamTinkerLaunch by downloading the script and running `./steamtinkerlaunch` to install it
2. You then ran `cd ~/stl/prefix` and ran `./steamtinkerlaunch hmm install nightly`?

<hr>

Unfortunately, you ran `steamtinkerlaunch --version` before attaching your log, so the log file is for that run of SteamTinkerLaunch and doesn't have any HMM related debugging information. I should've mentioned this -- sorry! :sweat: 

Could you rerun the HMM command and attach an updated log? Attaching it in a reply is fine :slightly_smiling_face: 

You can also run `./steamtinkerlaunch hmm uninstall` to remove the prefix and associated files for a clean reinstall once we get to the bottom of the issue.

P.S. - Just renaming the issue for my own clarity -- Your name is fine, but this will help me when browsing the issue tracker and keeping things organized

**Comment by McSkellyHo** [comment]
> Just to make sure:
You installed SteamTinkerLaunch by downloading the script and running ./steamtinkerlaunch to install it
You then ran cd ~/stl/prefix and ran ./steamtinkerlaunch hmm install nightly?

Yes, I made sure to open terminal from `/stl/prefix` but I didn't use `./steamtinkerlaunch hmm install nightly`, I used `./steamtinkerlaunch hmm start nightly` since that's all I've seen listed. I ran the install command just now.

[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10026497/steamtinkerlaunch.log)

**Comment by sonic2kk** [comment]
Ah sorry you are correct, my sleepy brain messed up there :-) That is correct usage though.

Gonna take a look at your logs, just wanted to say that you are correct there before I take a look.

**Comment by McSkellyHo** [comment]
Ah I see, don't worry about it. Though should I run `./steamtinkerlaunch hmm start nightly` and send the log of that? Because I ran the install command just now.

**Comment by sonic2kk** [comment]
It's fine, `hmm start` just runs the download and install commands :-)

Not sure what's causing the problem is *just* yet from reading your logs, but for some reason I suspect it's not actually trying to download the latest GE-Proton -- Though it says it wants to use the latest GE for some reason.

Definitely a bug, thanks for reporting!

**Comment by McSkellyHo** [comment]
No problem, thanks for helping out! I'm guessing I'll just have to wait for a patch?

**Comment by sonic2kk** [comment]
Could you go to `/dev/shm/steamtinkerlaunch` and open the `ProtonCSV.txt` file, and then look for `GE-Proton7-41`. Once you find it, the line should look like this:

```
GE-Proton7-41;/home/deck/.local/share/Steam/compatibilitytools.d/GE-Proton7-41/proton
```

If you can copy the path (that is the `/home/deck/.local/share/Steam/compatibilitytools.d/` part -- or whatever the path is, just leaving out the `/proton` at the end) and then try to go there in Dolphin, the KDE file explorer. You can right click on the path at the top and choose "edit", then paste it in and remove the `/proton` part at the end.

![image](https://user-images.githubusercontent.com/7917345/202320245-dac69d27-8277-47cc-9de3-cc9f0df28082.png)

> No problem, thanks for helping out! I'm guessing I'll just have to wait for a patch?

You could fix it by running `./steamtinkerlaunch settings`, and then at the bottom of the global menu (the "GLOBAL" button) you can change the Proton version, and then run `./steamtinkerlaunch hmm uninstall` and then try to reinstall.

I think this might just be a small SteamTinkerLaunch glitch that shouldn't have occured, not sure what caused it but I would be curious to know if a Proton version at that path does exist.

**Comment by McSkellyHo** [comment]
The text file showed it's in `/home/deck/.config/steamtinkerlaunch/proton/custom/GE-Proton7-41/` should I do something in this directory?
![image](https://user-images.githubusercontent.com/78847349/202321746-a3d4786b-f516-4ac9-8acf-7402665a9576.png)
I'll try to reinstall it right away.

**Comment by sonic2kk** [comment]
No that looks correct (that folder is where STL actually downloads its Proton versions :smile:) -- Very strange that STL doesn't see it from your logs...

Either way, a reinstall should fix it, and other users having issues have not reported this (and their logs seem to show it fine). Regardless, I will investigate and see if there are any code changes needed to make 110% sure the Proton version is set.

**Comment by McSkellyHo** [comment]
I switched to proton experimental in STL's global settings. But after the reinstall, the terminal output says HMM is already installed. 
(deck@steamdeck prefix)$ ./steamtinkerlaunch hmm start nightly
Preparing to install SteamTinkerLaunch on Steam Deck
Updating SteamTinkerLaunch to latest git

Dependency 'wget' already installed, nothing to do.
Dependency 'innoextract' already installed, nothing to do.
Dependency 'cabextract' already installed, nothing to do.
Dependency 'yad' already installed, nothing to do.

Finished updating SteamTinkerLaunch ('v11.12.20221115-5')!
Checking latest available HedgeModManager Development version
Latest HedgeModManager is already downloaded or was downloaded previously
Found up-to-date HedgeModManager executable - Nothing to download
Finished installing HedgeModManager
Adding HedgeModManager to application menu and setting up link handlers
Setting up installed HedgeModManager compatible games
Running configuration for 'Sonic Frontiers'
Finished configuring installed HedgeModManager games
Starting HedgeModManager

HMM still doesn't start

**Comment by sonic2kk** [comment]
Did you remember to run `./steamtinkerlaunch hmm uninstall`?

**Comment by McSkellyHo** [comment]
Yes, I did that first

**Comment by sonic2kk** [comment]
Hmm, I guess that command isn't working as expected. You'll have to manually delete the `hedgemodmanager` folder at `~/.config/steamtinkerlaunch` -- Not sure why `uninstall` isn't working though..

**Comment by sonic2kk** [comment]
Also, the terminal output is not very helpful :-) You should attach logs for each run instead, because re-reading your output it doesn't look like STL actually said HMM was already installed?

**Comment by McSkellyHo** [comment]
Oh sorry, I ran it again after manually deleting the HMM folder. But HMM still doesn't start.
[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10026666/steamtinkerlaunch.log)

**Comment by sonic2kk** [comment]
Thanks, this is much more useful! It looks like it's failing to install `dotnet48` -- Just like another issue that is currently open.

But since I have you here, can you check if you have a an `installDotNet.logHMM` file, or something named to that effect ending with `HMM`? You should have it now, since it can find the Proton version.

**Comment by sonic2kk** [comment]
Could you also delete the prefix again (`./steamtinkerlaunch hmm uninstall` should actually work, since it was actually uninstalled based on your terminal output), re-run the command, and attach the steamtinkerlaunch log and the installDotNet.logHMM file too? That way I can compare the one from the run you just did and the logs from the run I am asking you to do. It could have some useful hints :-)

**Comment by McSkellyHo** [comment]
Do you mean deleting the HMM folder like earlier? At the moment, there aren't any files ending with logHMM

**Comment by sonic2kk** [comment]
Hmm, I guess since the folder doesn't exist you can do the following. Sorry, it's late and maybe I'm not explaining the steps too well:
1. Go to `~/.config/steamtinkerlaunch` and delete the `hedgemodmanager` folder (I think that's what you meant, but I just wanted to be clear :slightly_smiling_face:)
2. Attempt to re-run `./steamtinkerlaunch hmm start nightly`
3. Once the output stops, attach the `steamtinkerlaunch.log` and, if one exists (which it should I believe), attach a file named `installDotNet.logHHM` or something with a similar file name that has `HMM` in the log title.

Doing this gives me two pieces of information:
1. More information on what is going on before STL attempts to set up the HMM Wineprefix
2. The other log file gives me information on what it is doing when it's in the process of setting up the prefix -- So I can see what the failures are in this process (I have a slight feeling it's going to complain about a lack of a wineserver -- but I'm not positive yet).

The issue you had initially, and this issue, may be connected. I have a feeling from looking at the code that STL may not be correctly setting `HMMWINE` -- Although it does actually catch it later on (even though the installation time is *far* too short to indicate success): `/home/deck/.local/share/Steam/steamapps/common/Proton - Experimental/files/bin/wine`

It could be that dotnet48 is already installed in that game prefix (I think it's the Sonic Frontiers prefix), but usually it completes a lot faster than that. I am interested in why it is failing to install dotnet48 on Steam Deck for HMM - A couple of users have reported this now but another user reported success. It could be that something broke and that in the step for configuring the HMM prefix, something is messed up and the Wine version isn't being caught / passed correctly. Which is why the `installDotNet.logHMM` file or a similar name will be useful to see :-)

I hope that helps give a bit of context. I realise being asked to follow seemingly arbitrary steps may be frustrating.

<hr>

It is getting late for me so I'll be signing off for tonight, but I'll try to get a chance to look at any logs tomorrow.

**Comment by McSkellyHo** [comment]
It is quite late for me as well. I got the two logs. The second one just doesn't have the HMM at the end.
[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10026774/steamtinkerlaunch.log)
[installDotNet.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10026776/installDotNet.log)

**Comment by sonic2kk** [comment]
Thanks, this was very interesting. The end of this dotnet log:

> warning: Note: command /home/deck/.local/share/Steam/steamapps/common/Proton - Experimental/files/bin/wine dotNetFx40_Full_x86_x64.exe /q /c:install.exe /q returned status 67. Aborting.

Is very interesting. I can't reproduce this error on my Steam Deck but I have seen this before mainly with out-of-date Winetricks (which the Steam Deck doesn't have).

I'll need to look into this tomorrow, but this could be the key here for the Steam Deck pain points. DotNet40 is failing to install for some reason, and it isn't even getting to attempting to install dotnet48 (Winetricks installs DotNet40 first, then dotnet48, as part of its dotnet48 installation).

Very very interesting. Thank you! I'll try to look into this asap

If you would like, when you have time you could try running `./steamtinkerlaunch hmm uninstall` and then in the global menu, selecting a GE-Proton version to run with. Then retry `./steamtinkerlaunch hmm start nightly`. Though I don't see this making much of a difference, it's possible that vanilla Proton has issues with the DotNet48 winetrick installation -- though I am skeptical of this.

Once I have more time I'll look into what could be causing the error. I noticed there were some registry complaints, so I'll have to get some time to look into this. rather busy at the moment but I'll do what I can.

Thanks for your efforts with helping debugging so far :-)

**Comment by McSkellyHo** [comment]
Oh I see, that's interesting. I'll have to continue with this later since it's 2 am now. Thank you for being so helpful!

**Comment by McSkellyHo** [comment]
I tried reinstalling HMM as you said, setting it to GE-Proton in STL's settings, though on the surface, nothing has changed, I'll send the two logs in case something changed in there.
[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10030819/steamtinkerlaunch.log)
[installDotNet.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10030820/installDotNet.log)

**Comment by sonic2kk** [comment]
The issue with this log is that HMMWINE is not being found.

I am not sure why you are getting the 67 error code with the other log, but I am hoping if I can resolve this HMMWINE issue, that will be a big step in the right direction.

**Comment by sonic2kk** [comment]
Can you go to `~/.config/steamtinkerlaunch` and send over your `global.conf` file? I wonder if this issue is springing from the global config not setting some variables correctly.

**Comment by McSkellyHo** [comment]
I have to send it in a ZIP since github can't send .conf files
[global.conf.zip](https://github.com/sonic2kk/steamtinkerlaunch/files/10036344/global.conf.zip)

**Comment by sonic2kk** [comment]
Ah yes -- Sorry about that, I forgot. Thank you though, the HMM var looks right so I'm not sure why I can't find it. *However*...

```
HMMCOMPDATA="/home/deck/.local/share/Steam/steamapps/common/SonicFrontiers"
```

This line is *certainly* not correct. Did you set this yourself? If not, this is quite alarming...

**Comment by McSkellyHo** [comment]
Oh! I did set that actually, I was trying to follow some half-finished guides before making an issue here and I set it to there as a guess.

**Comment by sonic2kk** [comment]
Ohhh okay. Well no worries :smile:  you can delete that line in your `global.conf` and STL will reset it back to the correct value.

I just pushed a commit that added a ton more logging to try and help debug this issue. This is not really intended to fix the issue but it will give more logging so I can trace through the issues more thoroughly.

If you can do the following for me:
1. Run `./steamtinkerlaunch` to update -- It should update to `20221117-3`
2. Run `./steamtinkerlaunch hmm uninstall` to remove HMM
3. Delete that `HMMCOMPDATA` line from your `global.conf`
4. Rerun `./steamtinkerlaunch hmm start nightly`
5. Reattach your `/dev/shm/steamtinkerlaunch/steamtinkerlaunch.log`

Basically every stage and every variable that passes through the function responsible for setting the Proton version is logged now, as well as logging at every stage of the `setHMMVars` function.

<hr>

Also as an aside - You are completely free to tinker as much as you want in SteamTinkerLaunch, but please remember that the tool is for experienced Linux users and the configuration options can have unexpected consequences if you don't know what you're doing, so just be careful when tinkering with the options :-)

**Comment by McSkellyHo** [comment]
[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10036459/steamtinkerlaunch.log)
installDotNet.logHMMPREFIX also appeared, but all it has inside is that `Warning: wineserver wasn't found.`

**Comment by sonic2kk** [comment]
>  installDotNet.logHMMPREFIX also appeared, but all it has inside is that `Warning: wineserver wasn't found.`

That is super helpful actually, thank you a lot.

I haven't checked the log just yet but this error comes from STL being unable to find the Proton version to run Winetricks with -- We're getting somewhere!

**Comment by McSkellyHo** [comment]
Should I try changing the proton version in stl's global settings and running `./steamtinkerlaunch hmm start nightly` again?

**Comment by sonic2kk** [comment]
You could try that but I don't know if it would make much difference -- I am not sure why it's giving these errors, so I guess it couldn't hurt to try :sweat_smile: 

It is very frustrating that I can't reproduce these errors but I am trying and investigating as best as I can with my limited time, thanks for the patience.

**Comment by sonic2kk** [comment]
AH! I was able to reproduce it!! The issue seems to be when it tries to use a Wine release in `~/.steamtinkerlaunch/proton/custom`! Now I need to figure out what's causing this...

Although in my case, it seems like it tries to continue installing - Though it gives a LOT of errors now..

Ah, it'll be continuing installation using my system wine

**Comment by McSkellyHo** [comment]
IT LAUNCHED
I set it to proton experimental and it finally launched!
It isn't dtecting Frontiers though

**Comment by sonic2kk** [comment]
You need to use the latest nightly for it to detect Frontiers -- What is the version at the top of your HMM window? If it isn't a hex string like `fe312fe" or something then it's probably not using the latest nightly. Hedge Mod Manager only has support for Sonic Frontiers in its nightly build at the moment.

Do you have any other Sonic games installed and is it seeing those?

Either way, I would recommend the following:
- Remove your Sonic Frontiers prefix, since you used that prefix before
- Do a full reinstall of HMM -- Run `./steamtinkerlaunch hmm uninstall` and then run `./steamtinkerlaunch hmm start`

The issue appears to be that STL can't find the correct version of Proton to use with the game if it tries to use GE-Proton from the STL custom download directory. That's a very strange problem and I'm not sure what's causing it.

**Comment by McSkellyHo** [comment]
The version shown is 7.8-2-793aff8, where is the Sonic Frontiers prefix exactly? I don't want to accidentally delete the wrong thing.

**Comment by McSkellyHo** [comment]
A little issue happened, I can't click on anything on my steam deck after I sent that comment, it's like everything is frozen except for the mouse cursor, I can't even put it to sleep mode, so I'll have to restart it.

Update: hmm still launches after the restart, the version shown is 7.8-2

**Comment by sonic2kk** [comment]
> where is the Sonic Frontiers prefix exactly? I don't want to accidentally delete the wrong thing

That depends on where Sonic Frontiers is installed.
- If it's on your main SSD, it's at `~/.local/share/Steam/steamapps/compatdata/1237320` - Delete the folder named `1237320`
- If it's on your microSD, it'll be at `/run/media/<some folder with mmc in the name>/steamapps/compatdata/1237320` - Delete the `1237320` folder
- It may be worth checking both of these paths and deleting the folder with that name, just in case it was created in the wrong place because of a Steam client bug

Once the prefix has been deleted, launch Sonic Frontiers once and close it again. 

> Update: hmm still launches after the restart, the version shown is 7.8-2

This means you did not configure HMM to use the nightly channel. Run `./steamtinkerlaunch settings` and then click the "Global" button, and at the very bottom under Release Channel, select "nightly". You can also make sure your Proton version is set to Experimental here (GE should work but it appears there is a bug at the moment which your efforts helped me uncover!). Then **Save** and close SteamTinkerLaunch

After this, **make sure you delete the Sonic Frontiers prefix before trying again**, and then run:
- `./steamtinkerlaunch hmm uninstall`
- `./steamtinkerlaunch hmm start` (it'll know to use the nightly because you configured it in the global menu)

After this, HMM should install using the correct Proton version. It should also find Sonic Frontiers.

Good luck!

**Comment by echometerain** [comment]
Migrating from #655, I contacted a friend uses HHM and they told me that I could not start HHM because I don't own any sonic games. Is this true or would there be some kind of error?

**Comment by sonic2kk** [comment]
That is probably an error, you should still be able to use HMM even if you don't own any Sonic games (in testing, originally HMM still started for me but it couldn't detect any games).

Also, please open a new issue. This issue is for a specific user on SteamOS, not a general HMM support issue :-)

**Comment by sonic2kk** [comment]
@McSkellyHo I just pushed a fix that forces a fallback to Proton Experimental (or Proton stable if it can't find experimental) -- This should work around the issue of Wine not being found. So just reset your prefixes and uninstall and reinstall HMM with STL.

**Comment by McSkellyHo** [comment]
It took a while because Frontiers needed to reinstall on steam, but I just finished the redownload with nightly and HMM is detecting Frontiers. Thank you so much for your help! And sorry about my own actions being cause of some issues from the compdata settings

**Comment by sonic2kk** [comment]
> And sorry about my own actions being cause of some issues from the compdata settings

Oh gosh, I'm really sorry if I came across the wrong way here! You're totally fine, with my other report I just wanted to point out that STL can break things if you fiddle too much. I noticed during testing tonight that the HMM compatdata can show up as ` (none)` which could be confusing, so maybe that caused it.

<hr>

> HMM is detecting Frontiers

*huge sigh of relief*

I will be honest, I have been kind of stressed trying to fix these issues. I know how it feels when stuff doesn't work so I've been staying up late trying to help troubleshoot issues and fix problems to get this working. I keep telling myself "it's a git release, there will always be bugs in in-dev software, users are trying to help" but I still get stressed thinking that users are there waiting for it to start working. And I am sure patience runs thin for users **in general** (not specifically you!)  when it's been several days and it feels like there's been no movement.

Well, I am honestly super glad that it is working. Good luck with your Frontiers modding, and I seriously mean this when I say thank you for your patience and your logs. Your troubleshooting helped me find a couple of underlying issues that I didn't catch initially in testing.

Boy, I have never been so satisfied when closing an issue haha. But please do feel free to either re-open or open another issue if HMM stops working. Most mods should work - one-click install currently doesn't at least on Steam Deck, so manual installation is the way to go for now. You may be able to right click on the 1-click GameBanana mod button -> copy link, and then in terminal run `xdg-open pasted-link` which should allow you to install the mod (however `rar` files will not work -- you'll need to download and extract those files manually).

**Comment by McSkellyHo** [comment]
Unfortunately another issue has risen up that Sonic Frontiers doesn't start anymore, neither from pressing save and play in HMM nor when launching from the steam client.

**Comment by sonic2kk** [comment]
Sounds like it didn't install dotnet48 for Sonic Frontiers... where is Sonic Frontiers installed? Is it on your microSD card?

**Comment by McSkellyHo** [comment]
No, it's all on the main SSD since I don't have a microSD card
/home/deck/.local/share/Steam/steamapps/common/SonicFrontiers/

**Comment by sonic2kk** [comment]
Hmm, it should definitely be working if it's on the main SSD.

The first thing to try: Reboot your Steam Deck and then try launching from Desktop Mode to see if that works.

If that fails: Try deleting your Sonic Frontiers prefix, then running `steamtinkerlaunch hmm uninstall` and then `steamtinkerlaunch hmm start` to reinstall, and send over your steamtinkerlaunch.log. I'll have a look to see if I can find any potential failures when its installing dotnet48.

It's possible that Proton Experimental is failing to install dotnet48 for the games. It's also possible that Sonic Frontiers needs a reboot before it'll launch for some reason. Not totally sure to be honest with you, but the log could help :-)

That won't get rid of any of your mods btw, they're stored in the game's folder :-)

**Comment by sonic2kk** [comment]
There is also a report that the latest version of the modloader broke Sonic Frontiers support, perhaps this is an issue external to HMM.

If you have any other games, you could try those (preferrably a 64bit one like Sonic Origins or something)

**Comment by sonic2kk** [comment]
GE-Proton may be required to actually install dotnet48 properly, Experimental and stable may be unable to do it, meaning I will need to investigate a fix...

**Comment by McSkellyHo** [comment]
So should I try reinstalling with GE-Proton? Because I tried reinstalling while keeping experimental and Frontiers still doesn't launch.
[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10037045/steamtinkerlaunch.log)

**Comment by sonic2kk** [comment]
> So should I try reinstalling with GE-Proton? Because I tried reinstalling while keeping experimental and Frontiers still doesn't launch.

I would try the following:

1. Go to `~/.config/steamtinkerlaunch/proton/custom` and delete `GE-Proton7-41`
2. Download the latest GE-Proton release manually using something like ProtonUp-Qt
3. Go into the SteamTinkerLaunch settings -> Global menu -> under Hedge Mod Manager settings (probably at the bottom), choose GE-Proton7-41, then save and quit
4. Run `./steamtinkerlaunch hmm uninstall`
5. Run `./steamtinkerlaunch hmm start`

I would also delete the Sonic Frontiers prefix again and relaunch -- Though Winetricks should be able to fix it without deleting, this is to be on the safe side.

**Comment by sonic2kk** [comment]
Unfortunately I'm out of time again for today, really sorry...

I will try and fix this over the weekend, but I may have to take a break soon...

**Comment by McSkellyHo** [comment]
It's incredibly late for me so I'll have to go for today as well, I'm waiting for it to reinstall right now so I'll just let you know if Frontiers launches or not and then I'll be off for tonight

**Comment by sonic2kk** [comment]
On my Steam Deck, GE-Proton seems to be having more success.

The issue seems to be that GE-Proton is not being used correctly, and so installation of the required Winetricks is failing.

At least I have something to go off of now to try and fix the issue.

HMM actually successfully installed on my Steam Deck with GE-Proton! Hopefully I can get these Wine issues sorted out eventually.

**Comment by McSkellyHo** [comment]
HMM installed with GE-Proton for me as well this time, but Sonic Frontiers still doesn't launch

**Comment by sonic2kk** [comment]
Woo! One problem down. Okay, so the issue is that HMM ***needs*** to absolutely be installed with HMM

A couple of things:
- Did you delete the Frontiers prefix before launching?
- Try launching with all mods disabled?
- Try launching with the modloader uninstalled (under the HMM Settings tab there's a button)
- Try forcing Sonic Frontiers to run with Proton Experimental from Steam (properties > force compat tool)

If it launches with the modloader uninstalled, then it's possibly a problem with dotnet48 not installing properly for the game.

Next time you have enough time to test, please delete the Frontiers' prefix, launch HMM and then send over the STL log file (STL will automatically retry installing dotnet48 for games on each launch and will stop if it's already installed - deleting the prefix will force a retry of installing dotnet48 and the log will let me see the time it took).

I feel like maybe it's trying to install dotnet48 with Experimental for Frontiers (and games in general).

<hr>

This has been really productive. I feel a bit better knowing the issue now and I'll try help you and fix the issues over the weekend all being well -- and then it's break time :-)

**Comment by McSkellyHo** [comment]
I did delete the Frontiers prefix before launching and I tried launching with all mods disabled and with proton experimental being set in Frontiers' properties on Steam. I'll test the rest tommorrow because I can barely concentrate.

**Comment by McSkellyHo** [comment]
The game launched when I had the modloader uninstalled. This is the log from when I launched HMM after deleting the Frontiers prefix.
[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10040543/steamtinkerlaunch.log)

**Comment by sonic2kk** [comment]
Thanks! I won't have time to investigate until after work but indeed dotnet48 was not installed for Sonic Frontiers:

> Fri Nov 18 11:44:38 AM CET 2022 SKIP - prepareHMMGames - Could not find compatdata dir for 'Sonic Frontiers' - Assuming that it is not installed

The checking of these prefixes may need to be less strict. Could you try running Sonic Frontiers once until the game screen opens, and then closing the game and then re-opening HMM? The check for the prefix may have been too strict and may not always set up prefixes for games that were never started before.

You can just start Frontiers from Desktop Mode and once the game window opens, you can quit the game. After this, re-open HMM and send over your STL log. Then I can have another look at the log. If opening HMM takes significantly longer this time you could assume dotnet48 is installed correctly.

Thanks, it may not seem like it but this is quite good progress! The main issues right now seem to be:
- STL doesn't always find the correct GE-Proton version to install HMM with, which would also explain some of the errors I've been seeing in the `installDotNet` log
- By extension, Winetricks *needs at least GE-Proton* to install dotnet48 correctly. Proton + Experimental can't do it (though once installed it *can* launch with these versions)
- STL doesn't seem to find Wineprefixes for games -- The check here may be too strict and could probably be loosened to not check for a prefix at all, and instead just create it if it doesn't exist.

Hoping that these won't be too hard to fix - STL downloads GE-Proton if not already downloaded but I guess the list of Proton versions doesn't update or contains bad data. That'll fix the first two issues, and the last one I'm *hoping* is just an oversight on my part and that launching the game once will resolve the issue.

Once you run Frontiers once, try reinstalling the modloader using the button on the HMM settings page and running your game. Good luck!

**Comment by McSkellyHo** [comment]
It took a little long on Running configuration for 'Sonic Frontiers' but not significantly longer. I still try launching the game with the modloader installed again.
[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10040680/steamtinkerlaunch.log)

**Comment by McSkellyHo** [comment]
So Frontiers still doesn't launch with the modloader installed

**Comment by sonic2kk** [comment]
That still looks like it's failing given the time between the dotnet logs (on my Steam Deck one install usually takes a little over 5 minutes but can sometimes take over 10 minutes, so 2 minutes is too shor):

```
Fri Nov 18 12:00:52 PM CET 2022 INFO - installDotNet - Starting dotnet48 install - check /dev/shm/steamtinkerlaunch/installDotNet.log
Fri Nov 18 12:00:52 PM CET 2022 INFO - installDotNet - WINEDEBUG="-all" WINEPREFIX="/home/deck/.local/share/Steam/steamapps/compatdata/1237320/pfx" WINE="/home/deck/.local/share/Steam/compatibilitytools.d/GE-Proton7-41/files/bin/wine" "/home/deck/.config/steamtinkerlaunch/downloads/winetricks/src/winetricks" --unattended "dotnet48"
Fri Nov 18 12:03:06 PM CET 2022 INFO - installDotNet - Stopped dotnet48 install - check /dev/shm/steamtinkerlaunch/installDotNet.log
Fri Nov 18 12:03:06 PM CET 2022 INFO - prepareHMMGames - Finished installing dotnet48 for 'Sonic Frontiers'
```

Can you send over any `installDotNet.log` files? I wonder why it's failing...

**Comment by McSkellyHo** [comment]
[installDotNet.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10040770/installDotNet.log)

**Comment by sonic2kk** [comment]
It actually looks like it succeeded - Maybe this is a problem recently introduced with Sonic Frontiers' modloader. I'll have to check later when I have some time.

Even though the time is too short, the dotnet log looks valid.  There's an open issue on the Hedge Mod Manager issue tracker that Sonic Frontiers is broken for two users, both on Windows. I'll have to check if this is the case.

Do you have any of the following games you could install and test with?
- Olympic Games Tokyo 2022
- Puyo Puyo Tetris 2
- Sonic Forces
- Sonic Origins

These are 64bit games (meaning dotnet48 will be installed for them too), and if these games work, that would rule out the possibility of this being a STL issue and could just be a general HMM issue at the moment (Sonic Frontiers support for HMM is still in ongoing development and is not in a stable release).

Since this is likely a modloader issue, and the modloader updates independently of HMM as far as I know, this issue probably can't be fixed by just rolling back to an older HMM version.

It unfortunately seems like this is a modloader issue external to STL/Linux, but I will test later and see for myself :-)

**Comment by McSkellyHo** [comment]
I have Sonic Forces so I'll install it and try running HMM with it. Playing Frontiers with mods feels so close yet so far lol. Thanks for all your help!

**Comment by sonic2kk** [comment]
I can reproduce the crashing on my end with Hedge Mod Manager and the modloader, and I personally verified that dotnet48 was installed.

This is a Frontiers modloader issue it seems like. Man I'm so sorry, but at least HMM is set up now, and you provided *amazing* feedback to help with debugging.

Related upstream issue: https://github.com/thesupersonic16/HedgeModManager/issues/494

<hr>

If Forces launches for you with its modloader installed (just a heads up - Make sure it's installed in the settings tab, and it wouldn't hurt to uninstall and reinstall just to be safe).

However Forces also uses the same modloader as Frontiers, so there is a chance it may not work either, and it may once again be unrelated to STL.

<hr>

I would say give this a bit more time until the HMM folks release an update for the modloader, and keep an eye on that upstream issue I linked for updates.

**Comment by McSkellyHo** [comment]
I just launched HMM after installing Forces and a window popped up about an update "fixed issue where save redirection were not working with mods on Sonic Frontiers"

**Comment by McSkellyHo** [comment]
Forces launched but Frontiers still doesn't

**Comment by sonic2kk** [comment]
If you want you can try enabling some codes for Forces, just to ensure the modloader is working.

If Forces loads codes/mods, then this is a Frontiers issue with HMM and probably not an STL bug. I know that doesn't make this any less frustrating for an end user but I guess at least you can know where the issue is coming from in that case? :sweat_smile: 

Thanks for all the testing.

**Comment by sonic2kk** [comment]
SteamTinkerLaunch v11.12.20221118-3 should have a fix in place for finding a valid GE-Proton version and will no longer fall back to Proton Experimental / Proton Stable.

Your efforts over the last almost 3 days were crucial in diagnosing the problems here. I'm just sorry that 1) it took so long to fix, and 2) that Frontiers has other issues :frowning: 

Installation of HMM will now be a lot more seamless on Steam Deck and other systems. Ubuntu still seems to have trouble.

Once you can verify that Forces correctly loads mods (something that would make it obvious like a model swap or infinite lives/whatever might be useful), feel free to close this issue.

Gosh, this was such a process but thanks for sticking with me. I'll be sure to document some of this on the wiki later as potential troubleshooting steps.

**Comment by McSkellyHo** [comment]
Mods work for Forces so it is an issue between HMM and Frontiers. I do admit, it is frustrating getting so far but Frontiers mods still being out of reach, but you've been extremely helpful, so thank you so much! At least I can play Forces as Sticks The Badger while waiting for an HMM update lol

**Comment by sonic2kk** [comment]
@McSkellyHo If you could try deleting the Frontiers' prefix, then running the game at least once, then opening HMM with STL, and then uninstalling and re-installing the modloader, could you report if that fixes Frontiers crashing?

**Comment by McSkellyHo** [comment]
The game still crashes before its window even opens

**Comment by sonic2kk** [comment]
Seems there may be some issues with Sonic Frontiers specifically with STL that I'll have to investigate (separate to this issue). Thanks for the re-test even though this has been closed for a while, I appreciate it :+1:
